DataFormats/ParticleFlowCandidate/src/classes_def.xml

```diff
@@ -24,7 +25,9 @@
   <ioread sourceClass = "reco::PFCandidate" version="[1-]" targetClass="reco::PFCandidate" source="std::vector<unsigned long long> refsInfo_" target="refsCollectionCache_">
     <![CDATA[refsCollectionCache_.clear();refsCollectionCache_.resize(onfile.refsInfo_.size(),0); ]]>
   </ioread>
-
+  <ioread sourceClass = "reco::PFCandidate" version="[-19]" targetClass="reco::PFCandidate" source="reco::PFCandidate::PFVertexType vertexType_" target="">
+    <![CDATA[newObj->setVertex(newObj->vertexLegacy(onfile.vertexType_));]]>
```

Comment:
@Dr15Jones @pcanal For my education, are there any ordering guarantees for the `ioread` rules? 

I was able to reproduce the exception, and it is thrown with the following stack trace
```
#1  0x00007ffff7f0e9d9 in edm::mustBeNonZero(edm::EDProductGetter const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, edm::ProductID const&) ()
   from .../libDataFormatsCommon.so
#2  0x00007fffcf189296 in reco::PFCandidate::trackRef() const ()
   from .../libDataFormatsParticleFlowCandidate.so
#3  0x00007fffcf17c64f in reco::PFCandidate::vertexLegacy(reco::PFCandidate::PFVertexType) const ()
   from .../libDataFormatsParticleFlowCandidate.so
#4  0x00007fffcf175bf7 in ROOT::read_recocLcLPFCandidate_3(char*, TVirtualObject*) ()
   from .../libDataFormatsParticleFlowCandidate.so
#5  0x00007ffff710a338 in int TStreamerInfo::ReadBufferArtificial<char**>(TBuffer&, char** const&, TStreamerElement*, int, int) ()
```
which looks to me that this rule would be called before the rule calling `edm::EDProductGetter::assignEDProductGetter()` (leaving `getter_ == nullptr` and the specific `Ref` constructor to throw).

(the question is mostly academical given that the subsequent de-reference of `Ref` would not work anyway)

Reply:
There is indeed an ordering.  The rules are supposed to be read after its input has been read and right after reading the output if it is persistent.  If no input is specified or if the output is not persistent then it is executed at the end. [If I recall correctly]